src/clients/opensea/index.ts

```diff
@@ -57,19 +63,33 @@ export class Opensea {
       throw new Error(`Invalid function name: ${functionName}`);
     }
 
-    const functionArgs = isMatchAdvancedOrders
-      ? [
-          transaction.input_data.orders,
-          transaction.input_data.criteriaResolvers,
-          transaction.input_data.fulfillments,
-          transaction.input_data.recipient,
-        ]
-      : [
-          transaction.input_data.advancedOrder,
-          transaction.input_data.criteriaResolvers,
-          transaction.input_data.fulfillerConduitKey,
-          transaction.input_data.recipient,
-        ];
+    let functionArgs: any[] = [];
+    if (isMatchAdvancedOrders) {
+      const inputData = transaction.input_data as InputDataMatchAdvancedOrders;
```

Comment:
no anys and casts here please

Reply:
You'll probably need https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates